### PR TITLE
feat: 🎸 Add procedure to apply all incoming balances for CA

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Description

Adds a procedure to apply incoming balances for a CA

### Breaking Changes

NA

### JIRA Link

DA-1087

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/utils/__tests__/conversion.ts [hunk 1]
```diff
@@ -2262,7 +2263,7 @@ describe('u128ToBigNumber', () => {
   });
 });
 
-describe('u16ToBigNumber', () => {
+describe('bigNumberToU16 anb u16ToBigNumber', () => {
```
Comment (polymath-eric) [inline src/utils/__tests__/conversion.ts:2266]:
```suggestion
describe('bigNumberToU16 and u16ToBigNumber', () => {
```

File: src/utils/conversion.ts [hunk 2]
```diff
@@ -1219,6 +1219,15 @@ export function meshPermissionsToPermissions(
   };
 }
 
+/**
+ * @hidden
+ */
+export function bigNumberToU16(value: BigNumber, context: Context): u16 {
+  assertIsInteger(value);
+  assertIsPositive(value);
```
Comment (polymath-eric) [inline src/utils/conversion.ts:1227]: What happens in the overflow case?

Does this need to assert its less than 2^16?

File: src/api/entities/confidential/ConfidentialAccount/types.ts [hunk 3]
```diff
@@ -34,3 +36,15 @@ export type ConfidentialAssetHistoryEntry = {
...
+export interface ApplyIncomingConfidentialAssetBalancesParams {
+  /**
+   * Confidential Account (or the public key of the ElGamal key pair) to which any incoming balance is to be applied
+   */
+  confidentialAccount: string | ConfidentialAccount;
+
+  /**
+   * The maximum number of incoming balances to apply
+   */
+  maxUpdates: BigNumber;
```
Comment (polymath-eric) [inline src/api/entities/confidential/ConfidentialAccount/types.ts:49]: What about making this optional and defaulting to all? 

I am not sure on the use case here. I figure I would want to apply all incoming balances most of the time though.

File: src/api/procedures/applyIncomingConfidentialAssetBalances.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  TxTags,
+} from '~/types';
+import { ExtrinsicParams, ProcedureAuthorization, TransactionSpec } from '~/types/internal';
+import { bigNumberToU16 } from '~/utils/conversion';
+import { asConfidentialAccount } from '~/utils/internal';
+
+/**
+ * @hidden
+ */
+export async function prepareApplyIncomingConfidentialAssetBalance(
```
Comment (polymath-eric) [inline src/api/procedures/applyIncomingConfidentialAssetBalances.ts:15]:
```suggestion
export async function prepareApplyIncomingConfidentialAssetBalances(
```

File: src/api/procedures/applyIncomingConfidentialAssetBalances.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      assets: [],
+      portfolios: [],
+    },
+  };
+}
+
+/**
+ * @hidden
+ */
+export const applyIncomingConfidentialAssetBalance = (): Procedure<
```
Comment (polymath-eric) [inline src/api/procedures/applyIncomingConfidentialAssetBalances.ts:85]:
```suggestion
export const applyIncomingConfidentialAssetBalances = (): Procedure<
```

File: src/utils/conversion.ts (hunk 2 above)
Comment (prashantasdeveloper) [reply src/utils/conversion.ts:1227]: Not sure, all others were also left as is

**Comment by prashantasdeveloper** [comment]
I'm merging this so that I can implement this in REST API, if there are any comments more here happy to implement them then @polymath-eric

**Pull request merged**